tests/testthat/helper-src.R

```diff
@@ -8,10 +8,10 @@ message("for ccheck_cardinality...()")
 d1 <- tibble::tibble(a = 1:5, b = letters[1:5])
 d2 <- tibble::tibble(a = c(1, 3:6), b = letters[1:5])
 d3 <- tibble::tibble(c = 1:5)
-d4 <- tibble::tibble(c = c(1:5, 5))
+d4 <- tibble::tibble(c = c(1:5, 5L))
 d5 <- tibble::tibble(a = 1:5)
 d6 <- tibble::tibble(c = 1:4)
-d7 <- tibble::tibble(c = c(1:5, 5, 6))
+d7 <- tibble::tibble(c = c(1:5, 5L, 6L))
```

Comment:
Can you tweak the test so that we check integer/numeric compatibility?

Reply:
Just so I got it right: we want a test that compares numeric columns with integer columns. I just checked, what `anti_join()` does locally:
``` r
suppressPackageStartupMessages(library(dplyr))
library(tibble)
t1 <- tibble(a = as.numeric(1:5))
t1
#> # A tibble: 5 x 1
#>       a
#>   <dbl>
#> 1     1
#> 2     2
#> 3     3
#> 4     4
#> 5     5
t2 <- tibble(a = 1:5)
t2
#> # A tibble: 5 x 1
#>       a
#>   <int>
#> 1     1
#> 2     2
#> 3     3
#> 4     4
#> 5     5
anti_join(t1, t2, by = "a")
#> # A tibble: 0 x 1
#> # … with 1 variable: a <dbl>
```

<sup>Created on 2019-09-23 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>
But on DB there might/will be differences. So is the test supposed to show the differences between DB- and local behaviour?

Reply:
I'd expect this behavior for all databases.